Bio/SeqIO/InsdcIO.py

```diff
@@ -743,42 +743,21 @@ def _write_the_first_line(self, record):
                 BiopythonWarning,
             )
 
-        # Get the base alphabet (underneath any Gapped or StopCodon encoding)
-        a = Alphabet._get_base_alphabet(record.seq.alphabet)
-        if not isinstance(a, Alphabet.Alphabet):
-            raise TypeError("Invalid alphabet")
-        elif isinstance(a, Alphabet.ProteinAlphabet):
-            units = "aa"
-        elif isinstance(a, Alphabet.NucleotideAlphabet):
-            units = "bp"
-        else:
-            # Must be something like NucleotideAlphabet or
-            # just the generic Alphabet (default for fasta files)
-            raise ValueError("Need a Nucleotide or Protein alphabet")
-
         # Get the molecule type
-        mol_type = self._get_annotation_str(record, "molecule_type", default=None)
+        mol_type = self._get_annotation_str(record, "molecule_type", default="DNA")
         if mol_type and len(mol_type) > 7:
             # Deal with common cases from EMBL to GenBank
             mol_type = mol_type.replace("unassigned ", "").replace("genomic ", "")
             if len(mol_type) > 7:
                 warnings.warn("Molecule type %r too long" % mol_type, BiopythonWarning)
-                mol_type = None
+                mol_type = "DNA"
         if mol_type in ["protein", "PROTEIN"]:
             mol_type = ""
 
-        if mol_type:
-            pass
-        elif isinstance(a, Alphabet.ProteinAlphabet):
-            mol_type = ""
-        elif isinstance(a, Alphabet.DNAAlphabet):
-            mol_type = "DNA"
-        elif isinstance(a, Alphabet.RNAAlphabet):
-            mol_type = "RNA"
+        if mol_type == "":
+            units = "aa"
         else:
-            # Must be something like NucleotideAlphabet or
-            # just the generic Alphabet (default for fasta files)
-            raise ValueError("Need a DNA, RNA or Protein alphabet")
+            units = "bp"
```

Comment:
There is also "rc" (record count) as used in WGA and other GenBank files which have no actual sequence.

Reply:
With Biopython release 1.77, the writer replaces "rc" in the input file "bp", at least for our test file in `Tests/GenBank`:

```
$ head -1 tls_KDHP01000000.gb 
LOCUS       KBUV01000000            3714 rc    DNA     linear   ENV 26-FEB-2018

$ python
Python 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import Bio
>>> from Bio import SeqIO
>>> Bio.__version__
'1.77'
>>> record = SeqIO.read('tls_KDHP01000000.gb', 'genbank')
>>> SeqIO.write([record], 'test.gb', 'genbank')
1
>>> ^D

$ head -1 test.gb 
LOCUS       KBUV01000000            3714 bp    DNA     linear   ENV 26-FEB-2018
```

Reply:
Sigh. That sounds like another subtle bug in the WGA support... they are not really sequences but are tantalisingly close to sequence based GenBank files.

Can you at least retain the ``raise ValueError("Need a DNA, RNA or Protein alphabet")`` behaviour when the molecule type is unknown? That should avoid having to tweak loads of tests.